test/behavior/undefined.zig

```diff
@@ -84,3 +84,51 @@ test "type name of undefined" {
     const x = undefined;
     try expect(mem.eql(u8, @typeName(@TypeOf(x)), "@TypeOf(undefined)"));
 }
+
+test "undefined propagation in equality operators for bool" {
+    if (builtin.zig_backend == .stage2_sparc64) return error.SkipZigTest; // TODO
+
+    var foo: bool = undefined;
+    const x = foo == undefined;
+    try expectEqual(x, @as(bool, undefined));
```

Comment:
You cannot test for undefined at runtime, these tests need to be compile error tests.

Reply:
> You cannot test for undefined at runtime, these tests need to be compile error tests.

Should I add all the tests from original issue? In the prev. pr I added test for just one of this. For me, it's quite unclear what was wrong with first pr. Can you explain this?

Reply:
Add something like this to `test/cases/compile_errors`:
```zig
test "undefined propagation in equality operators for bool" {
    var foo: bool = undefined;
    _ = foo == undefined;
    if (foo == undefined) {}
}

test "undefined propagation in equality operators for pointer" {
    var foo: *i32 = undefined;
    _ = foo == undefined;
    if (foo == undefined) {}
}

test "undefined propagation in equality operators for optional" {
    var foo: ?*i32 = undefined;
    _ = foo == undefined;
    if (foo == undefined) {}
}

test "undefined propagation in equality operators for error union" {
    var foo: anyerror = undefined;
    _ = foo == undefined;
    if (foo == @as(anyerror!i32, undefined)) {}
}

test "undefined propagation in equality operators for integers" {
    var foo: i32 = undefined;
    _ = foo == undefined;
    if (foo == undefined) {}
}

// error
// is_test=true
//
// :4:13: error: use of undefined value here causes undefined behavior
// :10:13: error: use of undefined value here causes undefined behavior
// :16:13: error: use of undefined value here causes undefined behavior
// :22:13: error: use of undefined value here causes undefined behavior
// :28:13: error: use of undefined value here causes undefined behavior
```

Sorry for not noticing this in the previous PR.

Reply:
+

> Add something like this to `test/cases/compile_errors`:
> 
> ```zig
> test "undefined propagation in equality operators for bool" {
>     var foo: bool = undefined;
>     _ = foo == undefined;
>     if (foo == undefined) {}
> }
> 
> test "undefined propagation in equality operators for pointer" {
>     var foo: *i32 = undefined;
>     _ = foo == undefined;
>     if (foo == undefined) {}
> }
> 
> test "undefined propagation in equality operators for optional" {
>     var foo: ?*i32 = undefined;
>     _ = foo == undefined;
>     if (foo == undefined) {}
> }
> 
> test "undefined propagation in equality operators for error union" {
>     var foo: anyerror = undefined;
>     _ = foo == undefined;
>     if (foo == @as(anyerror!i32, undefined)) {}
> }
> 
> test "undefined propagation in equality operators for integers" {
>     var foo: i32 = undefined;
>     _ = foo == undefined;
>     if (foo == undefined) {}
> }
> 
> // error
> // is_test=true
> //
> // :4:13: error: use of undefined value here causes undefined behavior
> // :10:13: error: use of undefined value here causes undefined behavior
> // :16:13: error: use of undefined value here causes undefined behavior
> // :22:13: error: use of undefined value here causes undefined behavior
> // :28:13: error: use of undefined value here causes undefined behavior
> ```
> 
> Sorry for not noticing this in the previous PR.

+